src/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQActive.cpp

```diff
@@ -16,15 +16,44 @@
   You should have received a copy of the GNU General Public License
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
+#include <iostream>
 
-#include <opm/parser/eclipse/Deck/UDAValue.hpp>
+#include <opm/io/eclipse/rst/state.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQActive.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQConfig.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQEnums.hpp>
-#include <iostream>
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
 
 namespace Opm {
 
+std::vector<UDQActive::RstRecord> UDQActive::load_rst(const UnitSystem& units,
+                                                      const UDQConfig& udq_config,
+                                                      const RestartIO::RstState& rst_state,
+                                                      const std::vector<std::string>& well_names,
+                                                      const std::vector<std::string>& group_names)
+{
+    std::vector<RstRecord> records;
+    const auto& rst_active = rst_state.udq_active;
+
+    for (const auto& record : rst_active.iuad) {
+        const auto& udq_input = udq_config[record.input_index];
+        UDAValue uda(udq_input.keyword(), units.uda_dim(record.control));
+        for (std::size_t use_index = 0; use_index < record.use_count; use_index++) {
+            std::size_t wg_index = rst_active.wg_index[record.wg_offset + use_index];
+
+            if (UDQ::well_control(record.control))
+                records.emplace_back(record.control, uda, well_names[wg_index]);
+            else {
+                if (UDQ::production_control)
```

Comment:
This line (`if (UDQ::production_control)`) gives a [warning](https://ci.opm-project.org/job/opm-common-PR-builder/4097/gcc/fileName.-16558701/source.c47caf56-ce55-4457-a508-cafb4ef66cbd/#47) that the "address of `UDQ::production_control` will never be `NULL`.  Did you mean to pass an argument here?

Reply:
Thank you - yes you are right this should be called with an argument.